akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorAdapter.scala

```diff
@@ -168,6 +174,22 @@ import scala.util.control.NonFatal
   }
 }
 
+@InternalApi
+private[typed] final class OptimizedActorAdapter[T](_initialBehavior: Behavior[T]) extends ActorAdapter[T](_initialBehavior) {
+  override def receive: Receive = null // not used
+
+  override protected[akka] def aroundReceive(receive: Receive, msg: Any): Unit = {
+    // as we know we never become in "normal" typed actors, it is just the behavior
+    // that changes, we can avoid the partial function/behavior stack entirely
+    // we also know that the receive should be total, so we can avoid the orElse part as well
+    msg match {
```

Comment:
do in the ActorAdapter and re-use running?

Reply:
The reason I introduced this is because ActorAdapter has another subclass which is the Guardian-wait-for-start one, and that interfered with any clever things I tried to do here.